src/node_process_methods.cc

```diff
@@ -272,6 +304,35 @@ void GetActiveHandles(const FunctionCallbackInfo<Value>& args) {
       Array::New(env->isolate(), handle_v.data(), handle_v.size()));
 }
 
+void GetActiveHandles2(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  std::vector<Local<Value>> handles;
+  for (HandleWrap* w : *env->handle_wrap_queue()) {
+    if (w->persistent().IsEmpty() || !HandleWrap::HasRef(w))
+      continue;
+
+    Local<Object> handle = Object::New(env->isolate());
+    if (handle->Set(env->context(),
+                    OneByteString(env->isolate(), "type"),
+                    OneByteString(env->isolate(), w->MemoryInfoName().c_str()))
+                        .IsNothing() ||
+        handle->Set(env->context(),
+                    OneByteString(env->isolate(), "asyncId"),
+                    Number::New(env->isolate(), w->get_async_id()))
+                        .IsNothing() ||
+        handle->Set(env->context(),
+                    OneByteString(env->isolate(), "triggerAsyncId"),
+                    Number::New(env->isolate(), w->get_trigger_async_id()))
+                        .IsNothing()) return;
```

Comment:
All of this is duplicated between both functions. Maybe put it in another function that's just for turning an AsyncWrap into that three-property JS object? It'd make for easier refactoring later, or adding an interface to get that for a single handle/request in the future.

Reply:
Thanks, I refactored the code accordingly